homeassistant/components/onvif/device.py

```diff
@@ -242,9 +242,8 @@ async def async_get_capabilities(self):
 
         pullpoint = False
         try:
-            event_service = self.device.create_events_service()
-            event_capabilities = await event_service.GetServiceCapabilities()
-            pullpoint = event_capabilities and event_capabilities.WSPullPointSupport
+            self.device.get_definition("pullpoint")
```

Comment:
This doesn't actually check for anything, because this service is defined as the `events` service, but that doesn't mean that the device actually supports PullPoint subscriptions. It's an optional implementation in the events service. Devices that adhere to the specification should expose support through the service capabilities call for the events service. If they do not, in my opinion it is not Home Assistant's responsibility to work around it.

I may be more amiable to an implementation that attempts to create the subscription and catches errors for cameras that do not support it in order to test capability, but this specifically will 100% break cameras without the functionality.

Reply:
You are right, of course. I copy the same check used for ptz functionality, but i didn't think that if ptz don't work, it doesn't broke anything, but if the pullpoint service is broken, it will broke the camera.
I made a fix to check if PullMessages operation work without exception, then set True the capabilities to actually listen the messages.
It should be a decent cross between don't have any check and check a service that the camera doesn't support. Of course, it would be better if the camera support it, but i cannot touch the camera code.